nodestream/project/project.py

```diff
@@ -181,13 +183,34 @@ def add_scope(self, scope: PipelineScope):
         Args:
             scope (PipelineScope): The scope to add.
         """
-        if self.plugins is not None:
-            for plugin in self.plugins:
-                if plugin.name == scope.name:
-                    scope.set_configuration(plugin.config)
-                    scope.set_targets(plugin.targets)
         self.scopes_by_name[scope.name] = scope
 
+    def add_plugin(self, plugin: PluginConfiguration):
+        """Adds a plugin to the project.
+
+        Args:
+            plugin (PluginConfiguration): The pluginto add.
+        """
+
+        self.plugins_by_name[plugin.name] = plugin
+
+    def add_plugin_pipeline_scope_from_resources(self, name: str, package: str):
+        """Adds a scope to the project.
+
+        Args:
+            scope (PipelineScope): The scope to add.
+        """
+        plugin_from_resources = PluginConfiguration.from_resources(
+            name=name, package=package
+        )
+        plugin_from_project_file = self.plugins_by_name[name]
+
+        plugin_from_resources.merge_pipeline_data(plugin_from_project_file)
+        scope = PipelineScope.from_file_data(
+            plugin_from_resources.name, plugin_from_resources.to_file_data()
+        )
```

Comment:
This is a little strange to me. I think `PluginConfiguration` instances should probably have a method called `make_scope` or something that knows how to build a `PipelineScope ` from itself.